### PR TITLE
LIB-214 - Update ncd and mental health forms to use javascript/css fo…

### PR DESCRIPTION
**Pull request by mseaton**
…r conditional condition-based logic rather than velocity expressions.

### CODE REVIEW

File: configuration/pih/htmlforms/mentalHealth.xml [hunk 1]
```diff
@@ -546,25 +546,6 @@
         </div>
     </section>
 
-
```
Comment (mseaton) [inline configuration/pih/htmlforms/mentalHealth.xml:549]: There is nothing here, as far as I can tell, so no need to preserve this logic.  Doing a search across the codebase for "mh-epilepsy-section" shows a section with this id in "epilepsy.xml", but not in this form.  If/when we want to add some conditional section in for epilepsy, we can follow the new patterns.

File: configuration/pih/htmlforms/mentalHealthIntake.xml [hunk 2]
```diff
@@ -211,9 +211,27 @@
         </style>
     </ifMode>
```
Comment (mseaton) [inline configuration/pih/htmlforms/mentalHealthIntake.xml:213]: Added the necessary javascript to support the commented out section below.  If uncommented, this should work in the same way as the NCD logic works on the ncd-adult-initial, just with a different concept set.

File: configuration/pih/htmlforms/ncd-adult-followup.xml [hunk 3]
```diff
@@ -8,20 +8,6 @@
     <postSubmissionAction class="org.openmrs.module.pihcore.htmlformentry.action.CleanDiagnosisConstructAction"/>
     <postSubmissionAction class="org.openmrs.module.pihcore.htmlformentry.action.CleanPrescriptionConstructAction"/>
 
-    <macros>
```
Comment (mseaton) [inline configuration/pih/htmlforms/ncd-adult-followup.xml:11]: The key change to this file is that everywhere that used to be wrapped in a macro conditional is now wrapped in an html element with a class.  Javascript queries the patient's current conditions as of the encounter date, and shows/hides sections for ncds based on this.

File: configuration/pih/htmlforms/ncd-adult-followup.xml [hunk 4]
```diff
@@ -394,8 +414,7 @@
             </div>
         </section>
```
Comment (mseaton) [inline configuration/pih/htmlforms/ncd-adult-followup.xml:416]: You will want to hide whitespace changes when viewing this diff.

File: configuration/pih/htmlforms/ncd-adult-initial.xml [hunk 5]
```diff
@@ -120,38 +131,6 @@
                 jq('#disposition select:first').change(function() {
                     jq('#disposition .field-error').first().html('').hide(); // hide any previous error if disposition is changed
                 });
-
```
Comment (mseaton) [inline configuration/pih/htmlforms/ncd-adult-initial.xml:123]: The below was made more generic and moved into the htmlFormFunctions class in the pihemr config

File: configuration/pih/scripts/global/liberia.js [hunk 6]
```diff
@@ -1,26 +1,14 @@
 
 
 var omrs = {
+
     toggleExposureDurationField: function (x, containerId) {
         if (x == 1) {
             $('#' + containerId).show();
         } else if (x == 0) {
             $('#' + containerId).hide();
         }
```
Comment (mseaton) [inline configuration/pih/scripts/global/liberia.js:10]: I think the best thing to do for now is to remove all of the Javascript below that was originally added (seemingly) to manipulate the conditions widget on the ncd and mental health forms.  If there are use cases that this was solving which can be identified and described, we can re-ticket to add back in the desired functionality.  Currently, it is not clear what (if anything) this is doing on the current forms, and it is causing JS errors on the console on all pages in the system outside of HFE edit/view mode, so I think it's best to remove for now.  @lnball ?

File: configuration/pih/htmlforms/mentalHealthIntake.xml [hunk 7]
```diff
@@ -211,9 +211,27 @@
...
+        function toggleConditionSections() {
+            jq('#mh-diagnoses').hide();
+            hfe.showSectionBasedOnCondition('PIH:14650', '#mh-diagnoses', false, true, true);
+        }
+
+        // When the form initially loads, and any time the encounter date is changed, show/hide sections based on condition list
+        jq('#encounterDate').find('input[type="hidden"]').change(function() {
+            toggleConditionSections()
+        });
+        toggleConditionSections();
```
Comment (mogoodrich) [inline configuration/pih/htmlforms/mentalHealthIntake.xml:230]: do we not need to wrap this in a jq(function())?  Granted, I've totally forgotten a lot of this, but doesn't that do something to make sure it doesn't execute until the DOM is fully mounted or something?

**Review by mogoodrich — APPROVED** [review]
Didn't deep super deep into that mammoth NCD Followup, but generally LGTM, thanks @mseaton !

File: configuration/pih/htmlforms/mentalHealthIntake.xml (hunk 7 above)
Comment (mseaton) [reply configuration/pih/htmlforms/mentalHealthIntake.xml]: Good point, done.

**Pull request merged**